src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -35,17 +36,22 @@ public class ClaimsTransformationMiddleware
 
         public async Task Invoke(HttpContext context)
         {
-            var handler = new ClaimsTransformationHandler(Options.Transformer, context);
+            var transform = Options.Transformer;
+            if (transform == null && Options.TransformerType != null)
+            {
+                transform = context.RequestServices.GetRequiredService(Options.TransformerType) as IClaimsTransformer;
```

Comment:
I know, I figure that's worth spending a bit of time thinking about today, since I don't want to check in anything right before the move, I'll play with what things look like if we nuked all the IFooEvents, kept the events types (as required base classes), and also introduce a sibling FooEventType similar to how this works (where we would use the type to GetService (if the instance is null, otherwise new up a FooEvent)